# Exploration of MAxO

The Medical Action Ontology (MAxO) (https://obofoundry.org/ontology/maxo) is an ontology of medical procedures, interventions, therapies, and treatments for disease with an emphasis on rare disease (RD).

This notebooks shows an exploration of the contents of MAxO. It focuses on the ontology content, but as
annotations are added we will include these.

## Setup

We will use the sqlite version of MAxO. In this notebook we will set up an alias for convenience.

Note on the command line the syntax is different, for unix systems:

```bash
alias maxo="runoak -i sqlite:obo:maxo"
```

In [21]:
%alias maxo runoak -i sqlite:obo:maxo

Let's start by getting some basic metadata about the ontology.

__NOTE__ the first time you run this there may be a lag while the pre-made sqlite database for maxo is downloaded

In [4]:
maxo ontology-metadata --all

dce:creator:
- Leigh Carmody
- Peter Robinson
dce:description:
- An ontology to represent medically relevant actions, procedures, therapies, interventions,
  and recommendations.
dce:title:
- Medical Action Ontology
dcterms:license:
- <https://creativecommons.org/licenses/by/4.0/>
id:
- obo:maxo.owl
owl:versionIRI:
- obo:maxo/releases/2022-12-19/maxo.owl
owl:versionInfo:
- '2022-12-19'
rdf:type:
- owl:Ontology
rdfs:isDefinedBy:
- http://purl.obolibrary.org/obo/obo.owl
schema:url:
- http://purl.obolibrary.org/obo/maxo.owl
sh:prefix:
- obo


## Root terms

We will query for roots (all terms without an is-a parent)

This reveals a few dangling classes that should be cleared up

In [5]:
maxo roots -p i

id	label
BFO:0000001	entity
CARO:0000003	None
CARO:0000006	None
CARO:0001001	None
CARO:0001010	None
CARO:0010000	None
CL:0000000	cell
CL:0017500	neutrophillic cytoplasm
CL:0017502	acidophilic cytoplasm
CL:0017503	basophilic cytoplasm
CL:0017504	polychromatophilic cytoplasm
CL:0017505	increased nucleus size
HP:0000001	All
MAXO:0000001	medical action
NCBITaxon:110815	None
NCBITaxon:147099	None
NCBITaxon:147554	None
NCBITaxon:189497	None
NCBITaxon:3176	None
NCBITaxon:3312	None
NCBITaxon:33630	None
NCBITaxon:33682	None
NCBITaxon:3378	None
NCBITaxon:38254	None
NCBITaxon:4891	None
NCBITaxon:4895	None
NCBITaxon:4896	None
NCBITaxon:4932	None
NCBITaxon:5782	None
NCBITaxon:Union_0000023	None
SO:0000110	sequence_feature


### Roots in the MAXO id space

As can be seen above there is a single root node in the MAXO namespace

We can also query for MAXO-roots, i.e classes in MAXO that have no MAXO is-a parent:

In [7]:
maxo roots -p i --has-prefix MAXO

## Visualization


We can use the `viz` command to see the full ancestry for any term or set of terms or term query.

For now, we will restrict to is-a parents (`-p i`):

In [5]:
maxo viz -p i 'biopsy of thymus' 'biopsy of thyroid gland' -o output/maxo-biopsy.png

![img](output/maxo-biopsy.png)

__TODO__: add explanation of why BFO processes show up

## Upper level

We can use the `tree` command to explore the upper level:



In [8]:
maxo tree -p i MAXO:0000001 MAXO:0000058 MAXO:0000002 MAXO:0000003 MAXO:0000013 MAXO:0000017 MAXO:0000021 MAXO:0001014

* [] **MAXO:0000001 ! medical action**
    * [i] **MAXO:0000002 ! therapeutic procedure**
        * [i] **MAXO:0000058 ! pharmacotherapy**
    * [i] **MAXO:0000003 ! diagnostic procedure**
    * [i] **MAXO:0000013 ! complementary and alternative medical therapy**
    * [i] **MAXO:0000017 ! preventative therapeutics**
    * [i] **MAXO:0000021 ! palliative care**
    * [i] **MAXO:0001014 ! medical action avoidance**


## Summary Statistics

We can get summary statistics using the `statistics` command.

Note that like many ontologies, MAXO comes "bundled" with other ontologies, so it's important to pass in
`--has-prefix` to restrict summary statistics only to things in the MAXO space.


In [22]:
maxo statistics --has-prefix MAXO > output/maxo-stats.yaml

This generates a YAML object (download from [output/maxo-stats.yaml](output/maxo-stats.yaml)).

let's take a look at the output. It's quite large so we'll use the unix `head` command:

In [23]:
!head -25 output/maxo-stats.yaml

annotation_property_count: 0
class_count: 1440
class_count_with_text_definitions: 1374
class_count_without_text_definitions: 66
contributor_summary:
  <http://orcid.org/0000-0001-7941-2961>:
    contributor_id: <http://orcid.org/0000-0001-7941-2961>
    role_counts:
      oio:created_by:
        facet: oio:created_by
        filtered_count: 822
  <http://orcid.org/0000-0002-7463-6306>:
    contributor_id: <http://orcid.org/0000-0002-7463-6306>
    role_counts:
      oio:created_by:
        facet: oio:created_by
        filtered_count: 3
  http://orcid.org/0000-0001-7941-2961:
    contributor_id: http://orcid.org/0000-0001-7941-2961
    contributor_name: http://orcid.org/0000-0001-7941-2961
    role_counts:
      oio:created_by:
        facet: oio:created_by
        filtered_count: 491
  http://orcid.org/0000-0001-7941-29610:


Note that the main stats are buried further down.


You can get the same thing in JSON. There is a cached copy you can view on https://jsonblob.com/1069350711447535616

In [24]:
maxo statistics --has-prefix MAXO -O json > output/maxo-stats.json

The YAML and JSON view is best for seeing the more complete granular results, but for many purposes a tabular view is better.

We can do that with `-O csv` (the `-O` is short for `--output-type`, which is available
for more OAK commands)

In [25]:
maxo statistics --has-prefix MAXO -O csv -o output/maxo-stats.tsv

Download: [output/maxo-stats.tsv](output/maxo-stats.tsv)

You can then browse the output using whatever method you prefer, e.g. load into excel.

In a Jupyter notebook, pandas dataframes are a convenient way to explore tabular data:

In [32]:
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [33]:
df = pd.read_csv("output/maxo-stats.tsv", sep="\t").fillna("")
df

id compared_with agents  class_count  deprecated_class_count  \
0  AllOntologies            []     []         1440                      53   

   non_deprecated_class_count  class_count_with_text_definitions  \
0                        1387                               1374   

   class_count_without_text_definitions  object_property_count  \
0                                    66                      4   

   annotation_property_count  named_individual_count  subset_count  \
0                          0                       0             0   

   rdf_triple_count  subclass_of_axiom_count  equivalent_classes_axiom_count  \
0             15642                     2842                             584   

  entailed_edge_count_by_predicate  distinct_synonym_count  \
0                               {}                    4391   

   synonym_statement_count class_count_by_category  \
0                     4391                      {}   

                                 contributor_summary change_summary  \
0  {'http://orcid.org/0000-0001-7941-2961': Contr...             {}   

   merged_class_query  deprecated_property_count  \
0                   0                          0   

   edge_count_by_predicate_BFO:0000051  edge_count_by_predicate_MAXO:0000521  \
0                                    8                                    20   

   edge_count_by_predicate_MAXO:0000864  edge_count_by_predicate_MAXO:0001015  \
0                                   134                                     9   

   edge_count_by_predicate_MAXO:0001027  edge_count_by_predicate_RO:0002233  \
0                                    76                                 441   

   edge_count_by_predicate_rdfs:subClassOf  \
0                                     2154   

   synonym_statement_count_by_predicate_hasBroadSynonym  \
0                                                 25      

   synonym_statement_count_by_predicate_hasExactSynonym  \
0                                               4250      

   synonym_statement_count_by_predicate_hasNarrowSynonym  \
0                                                 97       

   synonym_statement_count_by_predicate_hasRelatedSynonym  \
0                                                 19        

   mapping_statement_count_by_predicate_hasDbXref  \
0                                             554   

   mapping_statement_count_by_object_source_OGMS  \
0                                              8   

   mapping_statement_count_by_object_source_NCIT  \
0                                            330   

   mapping_statement_count_by_object_source_OAE  \
0                                             2   

   mapping_statement_count_by_object_source_SIO  \
0                                             1   

   mapping_statement_count_by_object_source_ERO  \
0                                             4   

   mapping_statement_count_by_object_source_NCRO  \
0                                              2   

   mapping_statement_count_by_object_source_SCDO  \
0                                              5   

   mapping_statement_count_by_object_source_OMIT  \
0                                              5   

   mapping_statement_count_by_object_source_CHEBI  \
0                                             134   

   mapping_statement_count_by_object_source_MeSH  \
0                                             20   

   mapping_statement_count_by_object_source_OBI  \
0                                            15   

   mapping_statement_count_by_object_source_IDO  \
0                                             1   

   mapping_statement_count_by_object_source_VO  \
0                                            4   

   mapping_statement_count_by_object_source_https  \
0                                               3   

   mapping_statement_count_by_object_source_PMID  \
0                                             11   

   mapping_statement_count_by_object_source_NCIt  \
0                    

the default TSV output for most OAK commands follows a TidyData format, with one row per item.

Let's unpivot this for easy viewing, using pandas `melt`:

In [34]:
df.melt()

variable  \
0                                                  id   
1                                       compared_with   
2                                              agents   
3                                         class_count   
4                              deprecated_class_count   
5                          non_deprecated_class_count   
6                   class_count_with_text_definitions   
7                class_count_without_text_definitions   
8                               object_property_count   
9                           annotation_property_count   
10                             named_individual_count   
11                                       subset_count   
12                                   rdf_triple_count   
13                            subclass_of_axiom_count   
14                     equivalent_classes_axiom_count   
15                   entailed_edge_count_by_predicate   
16                             distinct_synonym_count   
17                            synonym_statement_count   
18                            class_count_by_category   
19                                contributor_summary   
20                                     change_summary   
21                                 merged_class_query   
22                          deprecated_property_count   
23                edge_count_by_predicate_BFO:0000051   
24               edge_count_by_predicate_MAXO:0000521   
25               edge_count_by_predicate_MAXO:0000864   
26               edge_count_by_predicate_MAXO:0001015   
27               edge_count_by_predicate_MAXO:0001027   
28                 edge_count_by_predicate_RO:0002233   
29            edge_count_by_predicate_rdfs:subClassOf   
30  synonym_statement_count_by_predicate_hasBroadS...   
31  synonym_statement_count_by_predicate_hasExactS...   
32  synonym_statement_count_by_predicate_hasNarrow...   
33  synonym_statement_count_by_predicate_hasRelate...   
34     mapping_statement_count_by_predicate_hasDbXref   
35      mapping_statement_count_by_object_source_OGMS   
36      mapping_statement_count_by_object_source_NCIT   
37       mapping_statement_count_by_object_source_OAE   
38       mapping_statement_count_by_object_source_SIO   
39       mapping_statement_count_by_object_source_ERO   
40      mapping_statement_count_by_object_source_NCRO   
41      mapping_statement_count_by_object_source_SCDO   
42      mapping_statement_count_by_object_source_OMIT   
43     mapping_statement_count_by_object_source_CHEBI   
44      mapping_statement_count_by_object_source_MeSH   
45       mapping_statement_count_by_object_source_OBI   
46       mapping_statement_count_by_object_source_IDO   
47        mapping_statement_count_by_object_source_VO   
48     mapping_statement_count_by_object_source_https   
49      mapping_statement_count_by_object_source_PMID   
50      mapping_statement_count_by_object_source_NCIt   
51  mapping_statement_count_by_object_source_UMLS_CUI   
52       mapping_statement_count_by_object_source_CMO   
53  mapping_statement_count_by_object_source_NCBIBook   
54       mapping_statement_count_by_object_source_MMO   
55       mapping_statement_count_by_object_source_VSO   
56      mapping_statement_count_by_object_source_http   
57       mapping_statement_count_by_object_source_EFO   
58    mapping_statement_count_by_object_source_IDOMAL   
59  mapping_statement_count_subject_by_object_sour...   
60  mapping_statement_count_subject_by_object_sour...   
61  mapping_statement_count_subject_by_object_sour...   
62  mapping_statement_count_subject_by_object_sour...   
63  mapping_statement_count_subject_by_object_sour...   
64  mapping_statement_count_subject_by_object_sour...   
65  mapping_statement_count_subject_by_object_sour...   
66  mapping_statement_count_subject_by_object_sour...   
67  mapping_statement_count_subject_by_object_sour...   
68  mapping_statement_count_subject_by_object_sour...   
69  mapping_statement_count_subject_by_object_sour...  

Note that the original YAML report included many "nested" summaries, where counts were broken down by a facet - e.g. edge counts:

```yaml
edge_count_by_predicate:
  BFO:0000051:
    facet: BFO:0000051
    filtered_count: 8
  MAXO:0000521:
    facet: MAXO:0000521
    filtered_count: 20
  MAXO:0000864:
    facet: MAXO:0000864
    filtered_count: 134
...
```

These get flattened ("denormalized") in the conversion to TSV, creating a new column for each facet, but this isn't done for all objects since this would make too wide a table (e.g. the contributor report, some ontologies may have 100s of contributors).

E.g.

|variable|value|
|---|---|
|edge_count_by_predicate_BFO:0000051 | 8 | 
|edge_count_by_predicate_MAXO:0000521 | 20 |
|edge_count_by_predicate_MAXO:0000864 | 134 |

The fields names may be verbose but they are intended to be unambiguous. One problem with ontology stats reporting is that there is often built in ambiguity; e.g what does "number of terms" mean?

- number of classes
- number of classes plus relations (object properties?)
- are obsolete classes included? what about merged classes?

If you want to look up documentation for any of these fields, just prefix with `https://incatools.github.io/ontology-access-kit/datamodels/summary-statistics/`

E.g. https://incatools.github.io/ontology-access-kit/datamodels/summary-statistics/non_deprecated_class_count.html

(in future these elements may have their own w3id URIs)

Note this only works for "non-denormalized" fields - to interpret "edge_count_by_predicate_BFO:0000051" you must split this into the base statistic `edge_count_by_predicate` and the entity (BFO:0000051, aka has-part). Or you can simply look at the native normalized YAML output.

## Summarizing by branches

We can pass in a list of branches:

In [29]:
maxo statistics --has-prefix MAXO -O csv -o output/maxo-stats-roots.tsv MAXO:0000001 MAXO:0000058 MAXO:0000002 MAXO:0000003 MAXO:0000013 MAXO:0000017 MAXO:0000021 MAXO:0001014

You can download this: [output/maxo-stats-roots.tsv](output/maxo-stats-roots.tsv) and use excel etc to view.

Or see [this google sheet](https://docs.google.com/spreadsheets/d/1sKP7hhsY9YAVKEnLBqM6Zef_O80uOj6QPEvCuPYhrow/edit?usp=sharing)


In [37]:
df = pd.read_csv("output/maxo-stats-roots.tsv", sep="\t").fillna("")
df

id compared_with agents  \
0                                 medical action            []     []   
1                                pharmacotherapy            []     []   
2                          therapeutic procedure            []     []   
3                           diagnostic procedure            []     []   
4  complementary and alternative medical therapy            []     []   
5                      preventative therapeutics            []     []   
6                                palliative care            []     []   
7                       medical action avoidance            []     []   

   class_count  deprecated_class_count  non_deprecated_class_count  \
0         1387                       0                        1387   
1          256                       0                         256   
2          898                       0                         898   
3          556                       0                         556   
4            6                       0                           6   
5           22                       0                          22   
6            6                       0                           6   
7            9                       0                           9   

   class_count_with_text_definitions  class_count_without_text_definitions  \
0                               1374                                    13   
1                                248                                     8   
2                                885                                    13   
3                                556                                     0   
4                                  6                                     0   
5                                 22                                     0   
6                                  6                                     0   
7                                  9                                     0   

   object_property_count  annotation_property_count  named_individual_count  \
0                      0                          0                       0   
1                      0                          0                       0   
2                      0                          0                       0   
3                      0                          0                       0   
4                      0                          0                       0   
5                      0                          0                       0   
6                      0                          0                       0   
7                      0                          0                       0   

   subset_count  rdf_triple_count  subclass_of_axiom_count  \
0             0             15355                     2789   
1             0              3772                      556   
2             0             10555                     1853   
3             0              5888                     1140   
4             0                53                        6   
5             0               206                       34   
6             0                59                        8   
7             0                81                       18   

   equivalent_classes_axiom_count entailed_edge_count_by_predicate  \
0                             584                               {}   
1                             134                               {}   
2                             394                               {}   
3                             254                               {}   
4                               0                               {}   
5                               1                               {}   
6                               0                               {}   
7                               0                               {}   

   distinct_synonym_count  synonym_statement_count class_count_by_category  \
0                    4385                     4385                      {}

Here each row is a branch (note that numbers are cumulative, so numbers for a parent include
numbers for children, even if there is a separate row for that child).

Note that you will never see deprecated classes when using this option, if the deprecated classes are not in the hierarchy (as is conventional for OBO ontologies)

We may prefer to view the transposed form:

In [38]:
df.convert_dtypes().transpose()

0  \
id                                                                                     medical action   
compared_with                                                                                      []   
agents                                                                                             []   
class_count                                                                                      1387   
deprecated_class_count                                                                              0   
non_deprecated_class_count                                                                       1387   
class_count_with_text_definitions                                                                1374   
class_count_without_text_definitions                                                               13   
object_property_count                                                                               0   
annotation_property_count                                                                           0   
named_individual_count                                                                              0   
subset_count                                                                                        0   
rdf_triple_count                                                                                15355   
subclass_of_axiom_count                                                                          2789   
equivalent_classes_axiom_count                                                                    584   
entailed_edge_count_by_predicate                                                                   {}   
distinct_synonym_count                                                                           4385   
synonym_statement_count                                                                          4385   
class_count_by_category                                                                            {}   
contributor_summary                                 {'http://orcid.org/0000-0001-7941-2961': Contr...   
change_summary                                                                                     {}   
merged_class_query                                                                                  0   
deprecated_property_count                                                                           0   
edge_count_by_predicate_BFO:0000051                                                               8.0   
edge_count_by_predicate_MAXO:0000521                                                             20.0   
edge_count_by_predicate_MAXO:0000864                                                            134.0   
edge_count_by_predicate_MAXO:0001015                                                              9.0   
edge_count_by_predicate_MAXO:0001027                                                             76.0   
edge_count_by_predicate_RO:0002233                                                              441.0   
edge_count_by_predicate_rdfs:subClassOf                                                          2101   
synonym_statement_count_by_predicate_hasBroadSy...                                               25.0   
synonym_statement_count_by_predicate_hasExactSy...                                               4246   
synonym_statement_count_by_predicate_hasNarrowS...                                               97.0   
synonym_statement_count_by_predicate_hasRelated...                                               17.0   
mapping_statement_count_by_predicate_hasDbXref                                                  553.0   
mapping_statement_count_by_object_source_OGMS                                                     8.0   
mapping_statement_count_by_object_source_NCIT                                                   329.0   
mapping_statement_count_by_object_source_OAE                                                      2.0   
mappi

Note that the categories are not disjoint - some terms can be in more than one branch, and "medical action"
covers everything.


## Exploring mappings

Next let's use OAK to query mappings in MAXO. There are a lot, so for exploration purposes let's fetch this for all terms that have the string "biopsy" in them:

In [39]:
maxo mappings l~biopsy -O sssom

# curie_map: {}
# license: UNSPECIFIED
# mapping_set_id: temp
subject_id	subject_label	predicate_id	object_id	object_label	mapping_justification	subject_source	object_source
MAXO:0000332	biopsy of anus	oio:hasDbXref	NCIT:C51586		semapv:UnspecifiedMatching	MAXO	NCIT
MAXO:0001269	clinical biopsy	oio:hasDbXref	OBI:0002650	biopsy	semapv:UnspecifiedMatching	MAXO	OBI
MAXO:0001269	clinical biopsy	oio:hasDbXref	OBI:0002650	biopsy	semapv:UnspecifiedMatching	MAXO	OBI
MAXO:0001270	clinical computed tomography assisted biopsy	oio:hasDbXref	OBI:0002652	computed tomography assisted biopsy	semapv:UnspecifiedMatching	MAXO	OBI
MAXO:0001270	clinical computed tomography assisted biopsy	oio:hasDbXref	OBI:0002652	computed tomography assisted biopsy	semapv:UnspecifiedMatching	MAXO	OBI
MAXO:0001271	clinical core needle biopsy	oio:hasDbXref	OBI:0002656	core needle biopsy	semapv:UnspecifiedMatching	MAXO	OBI
MAXO:0001271	clinical core needle biopsy	oio:hasDbXref	OBI:0002656	core needle biopsy	semapv:Unspecified

In [40]:
maxo mappings l~biopsy -O sssom -o output/maxo-biopsy.sssom.tsv

In [41]:
df = pd.read_csv("output/maxo-biopsy.sssom.tsv", sep="\t", comment="#").fillna("")

In [42]:
df

subject_id                                 subject_label   predicate_id  \
0   MAXO:0000332                                biopsy of anus  oio:hasDbXref   
1   MAXO:0001269                               clinical biopsy  oio:hasDbXref   
2   MAXO:0001269                               clinical biopsy  oio:hasDbXref   
3   MAXO:0001270  clinical computed tomography assisted biopsy  oio:hasDbXref   
4   MAXO:0001270  clinical computed tomography assisted biopsy  oio:hasDbXref   
5   MAXO:0001271                   clinical core needle biopsy  oio:hasDbXref   
6   MAXO:0001271                   clinical core needle biopsy  oio:hasDbXref   
7   MAXO:0001272        clinical fine needle aspiration biopsy  oio:hasDbXref   
8   MAXO:0001272        clinical fine needle aspiration biopsy  oio:hasDbXref   
9   MAXO:0001273                  clinical image guided biopsy  oio:hasDbXref   
10  MAXO:0001273                  clinical image guided biopsy  oio:hasDbXref   
11  MAXO:0001274                        clinical needle biopsy  oio:hasDbXref   
12  MAXO:0001274                        clinical needle biopsy  oio:hasDbXref   
13  MAXO:0001275                      clinical surgical biopsy  oio:hasDbXref   
14  MAXO:0001275                      clinical surgical biopsy  oio:hasDbXref   
15  MAXO:0001276      clinical ultrasound-guided needle biopsy  oio:hasDbXref   
16  MAXO:0001276      clinical ultrasound-guided needle biopsy  oio:hasDbXref   

      object_id                         object_label  \
0   NCIT:C51586                                        
1   OBI:0002650                               biopsy   
2   OBI:0002650                               biopsy   
3   OBI:0002652  computed tomography assisted biopsy   
4   OBI:0002652  computed tomography assisted biopsy   
5   OBI:0002656                   core needle biopsy   
6   OBI:0002656                   core needle biopsy   
7   OBI:0002655        fine needle aspiration biopsy   
8   OBI:0002655        fine needle aspiration biopsy   
9   OBI:0002651                  image guided biopsy   
10  OBI:0002651                  image guided biopsy   
11  OBI:0002654                        needle biopsy   
12  OBI:0002654                        needle biopsy   
13  OBI:0002657                      surgical biopsy   
14  OBI:0002657                      surgical biopsy   
15  OBI:0002653      ultrasound-guided needle biopsy   
16  OBI:0002653      ultrasound-guided needle biopsy   

         mapping_justification subject_source object_source  
0   semapv:UnspecifiedMatching           MAXO          NCIT  
1   semapv:UnspecifiedMatching           MAXO           OBI  
2   semapv:UnspecifiedMatching           MAXO           OBI  
3   semapv:UnspecifiedMatching           MAXO           OBI  
4   semapv:UnspecifiedMatching           MAXO           OBI  
5   semapv:UnspecifiedMatching           MAXO           OBI  
6   semapv:UnspecifiedMatching           MAXO           OBI  
7   semapv:UnspecifiedMatching           MAXO           OBI  
8   semapv:UnspecifiedMatching           MAXO           OBI  
9   semapv:UnspecifiedMatching           MAXO           OBI  
10  semapv:UnspecifiedMatching           MAXO           OBI  
11  semapv:UnspecifiedMatching           MAXO           OBI  
12  semapv:UnspecifiedMatching           MAXO           OBI  
13  semapv:UnspecifiedMatching           MAXO           OBI  
14  semapv:UnspecifiedMatching           MAXO           OBI  
15  semapv:UnspecifiedMatching           MAXO           OBI  
16  semapv:UnspecifiedMatching           MAXO           OBI

In [44]:
maxo -a sqlite:obo:ncit lexmatch l~biopsy -o output/maxo-lexmatch-ncit-biopsy.sssom.tsv

In [45]:
df = pd.read_csv("output/maxo-lexmatch-ncit-biopsy.sssom.tsv", sep="\t", comment="#").fillna("")
df

subject_id                         subject_label     predicate_id  \
0   MAXO:0000332                        biopsy of anus  skos:closeMatch   
1   MAXO:0000332                        biopsy of anus  skos:closeMatch   
2   MAXO:0000336                 biopsy of bone marrow  skos:closeMatch   
3   MAXO:0000336                 biopsy of bone marrow  skos:closeMatch   
4   MAXO:0000336                 biopsy of bone marrow  skos:closeMatch   
5   MAXO:0000336                 biopsy of bone marrow  skos:closeMatch   
6   MAXO:0000338                       biopsy of brain  skos:closeMatch   
7   MAXO:0000338                       biopsy of brain  skos:closeMatch   
8   MAXO:0000338                       biopsy of brain  skos:closeMatch   
9   MAXO:0000340                      biopsy of breast  skos:closeMatch   
10  MAXO:0000340                      biopsy of breast  skos:closeMatch   
11  MAXO:0000340                      biopsy of breast  skos:closeMatch   
12  MAXO:0000341                    biopsy of bronchus  skos:closeMatch   
13  MAXO:0000350                       biopsy of colon  skos:closeMatch   
14  MAXO:0000350                       biopsy of colon  skos:closeMatch   
15  MAXO:0000355                    biopsy of duodenum  skos:closeMatch   
16  MAXO:0000355                    biopsy of duodenum  skos:closeMatch   
17  MAXO:0000359                   biopsy of esophagus  skos:closeMatch   
18  MAXO:0000366                       biopsy of heart  skos:closeMatch   
19  MAXO:0000366                       biopsy of heart  skos:closeMatch   
20  MAXO:0000373                      biopsy of larynx  skos:closeMatch   
21  MAXO:0000373                      biopsy of larynx  skos:closeMatch   
22  MAXO:0000375                         biopsy of lip  skos:closeMatch   
23  MAXO:0000375                         biopsy of lip  skos:closeMatch   
24  MAXO:0000376                       biopsy of liver  skos:closeMatch   
25  MAXO:0000376                       biopsy of liver  skos:closeMatch   
26  MAXO:0000376                       biopsy of liver  skos:closeMatch   
27  MAXO:0000380                        biopsy of lung  skos:closeMatch   
28  MAXO:0000380                        biopsy of lung  skos:closeMatch   
29  MAXO:0000380                        biopsy of lung  skos:closeMatch   
30  MAXO:0000380                        biopsy of lung  skos:closeMatch   
31  MAXO:0000381                  biopsy of lymph node  skos:closeMatch   
32  MAXO:0000381                  biopsy of lymph node  skos:closeMatch   
33  MAXO:0000381                  biopsy of lymph node  skos:closeMatch   
34  MAXO:0000385                       biopsy of mouth  skos:closeMatch   
35  MAXO:0000385                       biopsy of mouth  skos:closeMatch   
36  MAXO:0000385                       biopsy of mouth  skos:closeMatch   
37  MAXO:0000385                       biopsy of mouth  skos:closeMatch   
38  MAXO:0000386                      biopsy of muscle  skos:closeMatch   
39  MAXO:0000389                 biopsy of nasopharynx  skos:closeMatch   
40  MAXO:0000389                 biopsy of nasopharynx  skos:closeMatch   
41  MAXO:0000392                    biopsy of pancreas  skos:closeMatch   
42  MAXO:0000392                    biopsy of pancreas  skos:closeMatch   
43  MAXO:0000395                  biopsy of peritoneum  skos:closeMatch   
44  MAXO:0000395                  biopsy of peritoneum  skos:closeMatch   
45  MAXO:0000397                      biopsy of pleura  skos:closeMatch   
46  MAXO:0000398                      biopsy of rectum  skos:closeMatch   
47  MAXO:0000403             biopsy of small intestine  skos:closeMatch   
48  MAXO:0000403             biopsy of small intestine  skos:closeMatch   
49  MAXO:0000406                     biopsy of stomach  skos:closeMatch   
50  MAXO:0000409                      biopsy of testis  skos:closeMatch   
51  MAXO:0000409                      biopsy of testis  skos:closeMatch   
52  MAXO:0000411               